Fails to build: glibc 2.26 doesn't provide xlocale.h

silx 0.6.1 fails to build on Ubuntu 17.10 which includes glibc 2.26. glibc 2.26 no longer provides xlocale.h

You can see a build log at https://launchpad.net/ubuntu/+source/silx/0.6.1+dfsg-1. Click amd64.

Build log excerpt
-----------------------
silx/io/specfile/src/locale_management.c:27:10: fatal error: xlocale.h: No such file or directory
 #include <xlocale.h>
          ^~~~~~~~~~~
compilation terminated.

--- a/silx/io/specfile/src/locale_management.c
+++ b/silx/io/specfile/src/locale_management.c
@@ -24,15 +24,23 @@
 #include <stdlib.h>
 
 #ifdef _GNU_SOURCE
-#include <xlocale.h>
-#include <locale.h>
+#  include <locale.h>
+#  ifdef __GLIBC__
+#    include <features.h>
+#    if !((__GLIBC__ > 2) || ((__GLIBC__ == 2) && (__GLIBC_MINOR__ > 25)))
+#      /* strtod_l has been moved to stdlib.h since glibc 2.26 */
+#      include <xlocale.h>
+#    endif
+#  else 
+#    include <xlocale.h>
+#  endif
 #else
-#ifdef PYMCA_POSIX
-#else
-#ifdef SPECFILE_POSIX
-#include <locale.h>
-#endif
-#endif
+#  ifdef PYMCA_POSIX
+#  else
+#    ifdef SPECFILE_POSIX
+#      include <locale.h>
+#    endif
+#  endif
 #endif
 
 #include <string.h>

